### PR TITLE
Add CSharp and VB Project Configuration Properties for Automation

### LINKED ISSUE #1919: Create a new rule set file when CodeAnalysisRuleSet isn't defined at all
Related to #88.

When targeting the full framework, the `CodeAnalysisRuleSet` property is always defined. If it isn't set in the project, then the standard project imports set it to point to a file with some default set of FxCop rules turned on. When changing the severity of a rule through the diagnostic's node in Solution Explorer we simply bail out if the property isn't defined at all.

This doesn't work well with .NET Core projects, where there seems to be no default in the standard imports (and there's no good reason there should be). In this case, we should simply create a new, empty rule set file and update the project to use that.

We already do something similar if CodeAnalysisRuleSet points to a default file. Since we shouldn't modify those (they're part of the VS install) we instead copy the file, update the project to use the copy, and then modify it.

### LINKED ISSUE #2114: Changing the severity of a rule in the analyzers node doesnt take effect
1. Create a netstandard library
2. Add Microsoft.FxCopAnalyzers nuget package.
3. Make class1 derive from Attribute - this will fire CA1018.
4. Set CA1018's severity to None from solution explorer

Expected: the diagnostic would be suppressed
Found: A ruleset file is created but looks like that ruleset doesn't become the active one - the project file doesn't change to update it as the CodeAnalysisRuleSet property either.

### LINKED ISSUE #2137: Upgrade langversion codefix doesn't work for CPS projects
**Version Used**:  D15Rel IntPreview channel

**Steps to Reproduce**:

In a .NET Core project, write some code that uses a 7.1 feature like default literal.
I get a codefix for upgrading the project to use C# 7.1 or latest.
image

Invoking the codefix fail with this stack:

```
System.AggregateException : One or more errors occurred. ---> Unable to cast object of type 'Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectConfigProperties' to type 'VSLangProj80.CSharpProjectConfigurationProperties3'.
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.InvokeWorker(Func`1 getFromDocument,IProgressTracker progressTracker,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.<>c__DisplayClass18_0.<InvokeCore>b__0()
   at Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformAction(IExtensionManager extensionManager,Object extension,Action action)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
---> (Inner Exception #0) System.InvalidCastException : Unable to cast object of type 'Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectConfigProperties' to type 'VSLangProj80.CSharpProjectConfigurationProperties3'.
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.ApplyParseOptionsChanged(ProjectId projectId,ParseOptions options)
   at Microsoft.CodeAnalysis.Workspace.ApplyProjectChanges(ProjectChanges projectChanges)
   at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution,IProgressTracker progressTracker)
   at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.TryApplyChanges(Solution newSolution,IProgressTracker progressTracker)
   at Microsoft.CodeAnalysis.CodeActions.ApplyChangesOperation.TryApply(Workspace workspace,IProgressTracker progressTracker,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.CodeActions.CodeActionEditHandlerService.ProcessOperations(Workspace workspace,ImmutableArray`1 operations,IProgressTracker progressTracker,CancellationToken cancellationToken)
   at async Microsoft.CodeAnalysis.Editor.Implementation.CodeActions.CodeActionEditHandlerService.ApplyAsync(<Unknown Parameters>)<---

```

Looks like we are not implementing some legacy interface.

FYI @jcouv @jaredpar @Pilchie

### PR DESCRIPTION
**Pull request by basoundr**
Tagging @srivatsn @tmeschter @davkean @dotnet/project-system for review

**Customer scenario**

Add CSharp and VB Project Configuration Properties. These are the Project Configuration Properties which backs the Automation Objects obtained through DTE. We need this change to support 2 scenarios(As of now, we have 2. There will be more.)

1. Enable Codefixer to change the Language Version of a C# version to newer version = #2137 
2. Enable changing the severity of an analyzer rule from the Analyzer Node - #2114 

**Bugs this fixes:** 

Fixes #2137 Fixes #2114 Fixes #1919 

**Workarounds, if any**

None

**Risk**

Low - we are enabling a feature which did not work previously.

**Performance impact**

Low - since the new code does not deal with allocations or any of the known hot paths

**Is this a regression from a previous update?**

**Root cause analysis:**

This is a parity work to bring the new project system on par with the old project system and support back compat

**How was the bug found?**

Ad hoc testing

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Properties/CSharpProjectConfigurationProperties.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using VSLangProj110;
+using VSLangProj80;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [Export(ExportContractNames.VsTypes.ConfiguredProjectPropertiesAutomationObject)]
+    [Order(Order.Default)]
+    [AppliesTo(ProjectCapability.CSharp)]
+    public class CSharpProjectConfigurationProperties : AbstractProjectConfigurationProperties,
+        CSharpProjectConfigurationProperties3,
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Properties/CSharpProjectConfigurationProperties.cs]: 6 doesn't derive from 3?

Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Properties/CSharpProjectConfigurationProperties.cs]: 6 goes all the way still 4 and stops there. This is true for both VB and C#.
C# implements `CSharpProjectConfigurationProperties3` which implements `ProjectConfigurationProperties3`.
VB implements just `ProjectConfigurationProperties3` and there is no `VBProjectConfigurationProperties3`.

**Comment by basoundr** [comment]
@srivatsn Opened #2174

**Comment by basoundr** [comment]
Tagging @MattGertz for approval

**Comment by basoundr** [comment]
This will fix #1919 as well

**Comment by jcouv** [comment]
Thanks! 👍

**Comment by Neme12** [comment]
We need `AllowUnsafeBlocks` as well for this: https://github.com/dotnet/roslyn/pull/25875

**Pull request merged**